### PR TITLE
ARM template for scheduling start and stop operations on VIS using Azure Logic Apps

### PR DESCRIPTION
**Pull request by devanshjainms**
This PR includes of ARM template which will help users to deploy resource in their subscription, that will in turn help them to schedule start and stop operations on the Virtual Instance of SAP solutions and Database Instances of SAP.
The azure resources deployed for scheduling one operations are as follows:
- Action group: microsoft.insights/actionGroups
- Alert rule for start VIS
- Alert rule for stop VIS
- Alert rule for azure logic app action failure
- ARM API connection: connected to MSI given by users during the initial configuration
- Azure Logic App

**Action Group**: An **empty** action group is created where users can set up notification of their preferences.

**Alert rule for start VIS**: The scope of this set to all the VISs the users provides in the input parameter. The action group is also attached to this rule.

**Alert rule for stop VIS**: The scope of this set to all the VISs the users provides in the input parameter. The action group is also attached to this rule.

**Alert rule for azure logic app action failure**: This alert rule monitors the different actions in the logic app and triggers whenever there is a failure in the step. The action group is also attached to this rule.

**ARM API connection**: An API connector which is used to authenticate the ARM requests sent from the logic app. ARM connector is a managed connector and it connected to the MSI resource provided by the users during the initial configuration. The expectation is that this MSI should have access to perform start and stop operations on all the VISs listed.

**Azure Logic App**: A logic app is a azure resource which will be responsible for executing the automated workflows for start and stop VIS on a schedule. This logic app has primarily two primary actions, first is to start the VIS, and second is to stop the VIS. The logic app uses ARM connector in the logic apps to make ARM calls to execute the start and stop operations on the VIS and DBs.

### CODE REVIEW

File: ARMTemplates/SnoozeVIS/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,792 @@
...
+        "description": "Select start time to Power On SAP system",
+        "displayName": "Start Time",
+        "format": "timeFormat",
+        "required": false
+      }
+    },
+    "TimeZone": {
+      "type": "string",
+      "metadata": {
+        "description": "Select time zone for task schedule",
```
Comment (ajaygit158) [inline ARMTemplates/SnoozeVIS/azuredeploy.json:104]: Specify an example

File: ARMTemplates/SnoozeVIS/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,792 @@
...
+      "metadata": {
+        "description": "Remove the days in a week you want to skip the opeartion",
+        "required": false,
+        "displayName": "On these days"
+      }
+    },
+    "StartTime": {
+      "type": "string",
+      "metadata": {
+        "description": "Select start time to Power On SAP system",
```
Comment (ajaygit158) [inline ARMTemplates/SnoozeVIS/azuredeploy.json:95]: Specify an example

File: ARMTemplates/SnoozeVIS/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,792 @@
...
+        "format": "timeFormat",
+        "required": false
+      }
+    },
+    "TimeZone": {
+      "type": "string",
+      "metadata": {
+        "description": "Select time zone for task schedule",
+        "displayName": "Time zone",
+        "placeholder": "Start VM at this specific time zone",
```
Comment (ajaygit158) [inline ARMTemplates/SnoozeVIS/azuredeploy.json:106]: Should this be same as description?
Select time zone for task schedule

File: ARMTemplates/SnoozeVIS/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,792 @@
...
+      }
+    },
+    "VirtualInstanceForSAPSolutionsResourceID": {
+      "type": "array",
+      "metadata": {
+        "displayName": "Virtual Instance for SAP solutions Resource ID(s)",
+        "required": true
+      }
+    },
+    "Frequency": {
```
Comment (ajaygit158) [inline ARMTemplates/SnoozeVIS/azuredeploy.json:69]: These do not show up on UI

Comment (devanshjainms) [reply ARMTemplates/SnoozeVIS/azuredeploy.json:69]: If you are referring to At These Days/At these Hours, they are conditionally visible when certain "Frequency" is selected. Please let me know if you meant something else.

File: ARMTemplates/SnoozeVIS/azuredeploy.json (hunk 3 above)
Comment (devanshjainms) [reply ARMTemplates/SnoozeVIS/azuredeploy.json:106]: Did you mean the placeholder?

File: ARMTemplates/SnoozeVIS/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,792 @@
...
+        "displayName": "Perform operations in sequence",
+        "required": true
+      }
+    },
+    "SoftStopTimeout": {
+      "type": "int",
+      "defaultValue": 0,
+      "metadata": {
+        "required": false,
+        "displayName": "Soft stop timeout",
```
Comment (ajaygit158) [inline ARMTemplates/SnoozeVIS/azuredeploy.json:58]: SoftStop timeout in mins/seconds.. Please specify

**Review by ajaygit158 — APPROVED** [review]

**Pull request merged**